client/lib/promote-post/index.ts

```diff
@@ -71,6 +72,34 @@ export async function loadDSPWidgetJS(): Promise< void > {
 	await import( './string' );
 }
 
+const ANDROID_VERSION_HIDE_CAMPAIGNS_BUTTON = 22.9;
+
+const shouldHideGoToCampaignButton = () => {
+	// Android versions higher or equal than 22.9 should hide the button
+	// Trycatching it since user-agents can be changed to send custom data
+	try {
+		const userAgent = navigator.userAgent.toLowerCase();
+
+		// check if it's a wp-android device
+		if ( ! userAgent.includes( 'wp-android' ) ) {
+			return false;
+		}
+
+		// check if it's a version higher or equal than 22.9
+		const version = userAgent.split( '/' )[ 1 ];
```

Comment:
User agent's usually has other `/` on it, and it will make the value you are getting in the split incorrect.
For example, I am testing with this user-agent that Enej shared with me:
`Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wp-iphone/12.1`

We will need to update this to use regex, or if the `/` is always at the end, do a substring instead.